ge/org.osate.ge.errormodel/src/org/osate/ge/errormodel/ErrorBehaviorTransitionHandler.java

```diff
@@ -127,16 +128,21 @@ public boolean canDelete(final CanDeleteContext ctx) {
 	}
 
 	private BusinessObjectContext getSource(final BusinessObjectContext boc, final QueryService queryService) {
-		return queryService.getFirstResult(srcQuery, boc);
+		return queryService.getFirstBusinessObjectContextOrNull(srcQuery, boc);
 	}
 
 	private BusinessObjectContext getDestination(final BusinessObjectContext boc, final QueryService queryService) {
-		return queryService.getFirstResult(dstQuery, boc);
+		return queryService.getFirstBusinessObjectContextOrNull(dstQuery, boc);
 	}
 
 	@Override
 	public String getName(final GetNameContext ctx) {
-		return ctx.getBusinessObject(ErrorBehaviorTransition.class).map(bo -> {
+		return ctx.getBusinessObject(ErrorBehaviorTransition.class).map(bo -> bo.getName()).orElse("");
+	}
+
+	@Override
+	public String getNameForDiagram(final GetNameForDiagramContext ctx) {
+		return ctx.getBusinessObjectContext().getBusinessObject(ErrorBehaviorTransition.class).map(bo -> {
 			final String name = bo.getName() == null ? "" : bo.getName() + ": ";
 			return name + getText(bo.getCondition()).map(t -> "-[" + t + "]").orElse("");
```

Comment:
Name could be "-[ + t + "]->".  Since the beginning "-" is there, I think the ending arrow makes sense to have, unless there is a reason it's not there.

Also a thought, should we support changing the Event for a behavior transition from the Properties View?

Reply:
Good point regarding the inconsistency. After additional consideration, I've decided to remove the condition entirely. The reason for this is that the user is likely to want to display the condition but not the name. At this time we will not support showing the condition at all. It will be available via the tooltip. 

We will not support adjusting the event at this time via the properties view. It is out of scope for this issue. The intention with regards to the state machine editing is to get the support to the point where we can release it and it won't crash and won't cause us problems when expanding support later.